### PR TITLE
Write some new unit tests for expandybird

### PR DESCRIPTION
**Pull request by sparkprime**

### CODE REVIEW

**Comment by sparkprime** [comment]
@jackgr

File: cmd/expandybird/expander/expander_test.go [hunk 1]
```diff
@@ -16,168 +16,664 @@ limitations under the License.
...
 
-func (etc *ExpanderTestCase) GetTemplate(t *testing.T) *common.Template {
-	template, err := util.NewTemplateFromFileNames(etc.TemplateFileName, etc.ImportFileNames)
-	if err != nil {
-		t.Fatalf("cannot create template for test case '%s': %s", etc.Description, err)
-	}
+// content provides an easy way to provide file content verbatim in tests.
+func content(lines []string) []byte {
+	return []byte(strings.Join(lines, "\n") + "\n")
+}
```
Comment (jackgr) [inline cmd/expandybird/expander/expander_test.go]: You can also do this with literals, which is common and may be more readable, like this:

```go
var data []byte(`
    foo:
        foo1:bar1
        foo2:bar2
    bar: baz
`)

```

Comment (sparkprime) [inline cmd/expandybird/expander/expander_test.go]: Unfortunately the indentation there messes up Python and YAML

**Comment by sparkprime** [comment]
OK I just rebased, enabled the Python multifile test and added another test for the case to test failed python imports (file not in the chart).  PTAL

File: cmd/expandybird/expander/expander_test.go (hunk 1 above)
Comment (jackgr) [inline cmd/expandybird/expander/expander_test.go]: Really? I've used it a ton for YAML w/no problem. Indentation is not required, nor enforced by go fmt. It's just my example that indented. Fixed that now.

**Comment by jackgr** [comment]
Any chance of following the best practice wrt inline content?

**Comment by sparkprime** [comment]
The go indent is hard tabs, which is not allowed in YAML code.  Unless I'm missing something?

**Comment by sparkprime** [comment]
And Python gives
```
IndentationError: unexpected indent
```

File: cmd/expandybird/expander/expander_test.go (hunk 1 above)
Comment (sparkprime) [inline cmd/expandybird/expander/expander_test.go]: Just seen this -- yeah this is possible but it disrupts the flow of the file a lot when you have many more levels of indenting (7 in this case).  I've seen some Python code use text.dedent to strip it off instead.  Jsonnet and YAML have the ||| and | operators respectively for avoiding this problem.

**Comment by jackgr** [comment]
So, check out https://github.com/kubernetes/kubernetes/blob/master/pkg/util/strategicpatch/patch_test.go#L251. Wonder what's different? Could it be the marshaling?

**Pull request merged**